modules/styling/stories/Stencils.stories.mdx

```diff
@@ -0,0 +1,270 @@
+component, a stencil should be created for each subcomponet. If your component is a config
+component, a stencil can have nested styles.
+
+<SymbolDoc name="createStencil" />
+
+## Concepts
+
+### Base styles
+
+Base styles are always applied to a Stencil. All your default styles should go here. Base styles
+support psuedo selectors like `:focus-visible` or `:hover` as well as child selectors. Any selector
+supported by `@emotion/css` is valid here. All styles must be static and statically analyzable by
+the tranformer. If you need dynamic styling, look at Variables and Modifiers.
+
+### Variables
+
+Variables allow some properties to be dynamic. They work by creating CSS variables and are applied
+using the [style](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/style) property of an
+element to locally scope an override.
+
+Here's a simplified example:
+
+```tsx
+const myStencil = createStencil({
+  vars: {
+    defaultColor: 'red' // default value
+  },
+  base: ({defaultColor}) => {
+    color: defaultColor // `defaultColor` is '--abc123-defaultColor', not 'red'
+  }
+})
+
+const elemProps = myStencil({color: 'blue'}) // {style: {'--abc123-defaultColor': 'blue'}}
+
+<div {...elemProps} />
+```
+
+This will produce the following HTML:
+
+```html
+<style>
+  .css-abc123 {
+    --abc123-defaultColor: red;
+    color: var(--abc123-defaultColor);
+  }
+</style>
+<div class="css-123abc" style="--abc123-defaultColor: blue;"></div>
+```
+
+The element will have a `color` property of `'blue'` because the element style is the highest
+specificity and wins over a local class name. In the "Styles" tab of developer tools, it will look
+like the following:
+
+```
+element.style {
+  --abc123-defaultColor: blue;
+}
+
+.css-abc123 {
+  --abc123-defaultColor: red;
+  color: var(--abc123-defaultColor); // blue
+}
+```
+
+Variables are automatically added to the config of a Stencil. They share the same namespace as
+modifiers, so do not have a modifier with the same name as a variable.
+
+**Note:** Variables should be used sparingly. Style properties can be easily overridden without
+variables. Variables are useful if you want to expose changing properties regardless of selectors.
+For example, Buttons use a variables for colors of all states (hover, active, focus, disabled, and
+nested icons). Without variables, overriding the focus color would require deeply nested selector
+overrides.
+
+### Modifiers
+
+Modifiers are modifications to base styles. It should be used to change the appearance of a base
+style. For example, a button may have a modifier for "primary" or "secondary" which may change the
+visual emphasis of the button. Each modifier has its own CSS class name and the stencil will return
+the correct CSS classes to apply to an element based on what modifiers are active.
+
+```tsx
+const buttonStencil = createStencil({
+  base: {
+    padding: 5
+    // base styles
```

Comment:
```suggestion
     background: 'white',
    // base styles
```
In the above codeblock you mention that modifiers are used to change some base style, i think adding background here, shows how when a modifier is active, it will change the base style, in this case the background color

Reply:
If the modifier is optional, a base property value might make sense. If a modifier is not optional, there's no point in having a property that will always be overridden. It just takes more bytes.